src/evict/evict_lru.c

```diff
@@ -2126,23 +2133,28 @@ __evict_walk_tree(WT_SESSION_IMPL *session, WT_EVICT_QUEUE *queue, u_int max_ent
         if (__wt_cache_aggressive(session))
             goto fast;
 
-        /*
-         * If the global transaction state hasn't changed since the last time we tried eviction,
-         * it's unlikely we can make progress. This heuristic avoids repeated attempts to evict the
-         * same page.
-         */
-        if (!__wt_page_evict_retry(session, page)) {
-            WT_STAT_CONN_INCR(session, cache_eviction_server_skip_pages_retry);
-            continue;
-        } else if (modified &&
-          page->modify->update_txn >= __wt_atomic_loadv64(&conn->txn_global.last_running)) {
+        if (modified) {
             /*
-             * FIXME-WT-11805: The assumption that the eviction will fail if most recent update on
-             * the page from the transaction that is greater than the last running transaction has
-             * changed because now eviction also has it's own snapshot for visibility check.
+             * If the global transaction state hasn't changed since the last time we tried eviction,
+             * it's unlikely we can make progress. This heuristic avoids repeated attempts to evict
+             * the same page.
              */
-            WT_STAT_CONN_INCR(session, cache_eviction_server_skip_pages_last_running);
-            continue;
+            if (!__wt_page_evict_retry(session, page)) {
+                WT_STAT_CONN_INCR(session, cache_eviction_server_skip_pages_retry);
+                continue;
+            }
+
+            /*
+             * If we are under cache pressure, allow evicting pages with newly committed updates to
+             * free space. Otherwise, avoid doing that as it may thrash the cache.
+             */
+            if (F_ISSET(cache, WT_CACHE_EVICT_HARD)) {
+                if (!__txn_visible_id(session, page->modify->update_txn))
+                    continue;
+            } else if (page->modify->update_txn >= conn->txn_global.last_running) {
+                WT_STAT_CONN_INCR(session, cache_eviction_server_skip_pages_last_running);
+                continue;
+            }
```

Comment:
A more aggressive strategy can do:

```
            /*
             * If we are under cache pressure, allow evicting pages with newly committed updates to
             * free space. Otherwise, avoid doing that as it may thrash the cache.
             */
            if (F_ISSET(cache, WT_CACHE_EVICT_HARD)) {
                if (!__txn_visible_id(session, page->modify->update_txn))
                    continue;
            } else {
                if (!__wt_page_evict_retry(session, page)) {
                    WT_STAT_CONN_INCR(session, cache_eviction_server_skip_pages_retry);
                    continue;
                }

                if (page->modify->update_txn >= conn->txn_global.last_running) {
                     WT_STAT_CONN_INCR(session, cache_eviction_server_skip_pages_last_running);
                     continue;
                } 
            }
```